gp-includes/url.php

```diff
@@ -146,14 +146,14 @@ function gp_url_current() {
  * @return string
  */
 function gp_url_project( $project_or_path = '', $path = '', $query = null ) {
-	$project_path = urlencode( is_object( $project_or_path ) ? $project_or_path->path : $project_or_path );
+	$project_path = is_object( $project_or_path ) ? $project_or_path->path : $project_or_path;
 
 	// A leading double-slash will avoid prepending /projects/ to the path.
 	// This was introduced to enable linking to the locale glossary.
 	if ( '//' === substr( $project_path, 0, 2 ) ) {
-		$project_path = ltrim( $project_path, '/' );
+		$project_path = urlencode( ltrim( $project_path, '/' ) );
 	} else {
-		$project_path = array( 'projects', $project_path );
+		$project_path = array( 'projects', urlencode( $project_path ) );
 	}
 
```

Comment:
Let's just urlencode once, after the if condition, instead of once in each part of the if.

Reply:
How do you propose that @toolstack? The $project_path can be either a string or an array after the if statement and `urlencode` only supports strings.

Reply:
That's a good point :smile: